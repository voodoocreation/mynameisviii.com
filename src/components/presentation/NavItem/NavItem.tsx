import * as React from "react";

import Link from "../Link/Link";

interface IProps {
  href?: string;
  isSelected?: boolean;
  onClick?: () => void;
  params?: {};
  prefetch?: boolean;
  route?: string;
}

const NavItem: React.SFC<IProps> = ({
  children,
  isSelected,
  onClick,
  ...linkProps
}) =>
  isSelected ? (
    <li className="isSelected" onClick={onClick}>
      {children}
    </li>
  ) : (
    <li onClick={onClick}>
      <Link {...linkProps}>{children}</Link>
    </li>
  );

NavItem.defaultProps = {
  isSelected: false,
  onClick: () => undefined
};

export default NavItem;
