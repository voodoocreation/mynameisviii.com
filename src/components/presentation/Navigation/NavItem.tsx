import * as React from "react";
import { FormattedMessage } from "react-intl";

import Link from "../Link/Link";

interface IProps {
  href?: string;
  isSelected: boolean;
  messageId: string;
  onClick?: () => void;
  params?: {};
  prefetch?: boolean;
  route?: string;
}

const NavItem: React.SFC<IProps> = ({
  isSelected,
  messageId,
  onClick,
  ...linkProps
}) =>
  isSelected ? (
    <li className="isSelected" onClick={onClick}>
      <FormattedMessage id={messageId} />
    </li>
  ) : (
    <li onClick={onClick}>
      <Link {...linkProps}>
        <FormattedMessage id={messageId} />
      </Link>
    </li>
  );

NavItem.defaultProps = {
  onClick: () => undefined
};

export default NavItem;
