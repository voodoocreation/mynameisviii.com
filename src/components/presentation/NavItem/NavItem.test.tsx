import { mount, render } from "enzyme";
import * as React from "react";

import NavItem from "./NavItem";

const setup = (fn: any, fromTestProps?: any) => {
  const props = {
    ...fromTestProps
  };

  return {
    actual: fn(<NavItem {...props}>Test nav item</NavItem>),
    props
  };
};

describe("[containers] <NavItem />", () => {
  it("renders correctly with minimum props", () => {
    const { actual } = setup(render);
    expect(actual).toMatchSnapshot();
  });

  it("renders correctly when isSelected=true", () => {
    const { actual } = setup(render, { isSelected: true });
    expect(actual).toMatchSnapshot();
  });

  it("triggers `onClick` prop when item is clicked on", () => {
    const { actual, props } = setup(mount, { onClick: jest.fn() });

    actual.simulate("click");
    expect(props.onClick).toHaveBeenCalled();
  });

  it("doesn't throw an error when `onClick` prop is undefined and item is clicked on", () => {
    let isPassing = true;
    const { actual } = setup(mount);

    try {
      actual.simulate("click");
    } catch (error) {
      isPassing = false;
    }

    expect(isPassing).toBe(true);
  });
});
