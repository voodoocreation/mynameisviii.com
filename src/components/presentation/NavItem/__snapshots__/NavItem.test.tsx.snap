// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[containers] <NavItem /> renders correctly when isSelected=true 1`] = `
<li
  class="isSelected"
>
  Test nav item
</li>
`;

exports[`[containers] <NavItem /> renders correctly with minimum props 1`] = `
<li>
  <span>
    Test nav item
  </span>
</li>
`;
